pages/_conferenceCode/contacts/index.vue

```diff
@@ -346,48 +360,120 @@ function initData (){
-    _this.filter.selectedOrganizationType ||
-    _this.filter.selectedOrganization.length ||
-    _this.filter.selectedCountry.length ||
-    _this.filter.selectedFlag.length ||
-    _this.filter.selectedAttendance.length ||
-    _this.filter.selectedScop
+    this.filter.name ||
+    this.filter.isBroadSearch ||
+    this.filter.selectedMeetings.length ||
+    this.filter.selectedOrganizationType ||
+    this.filter.selectedOrganization.length ||
+    this.filter.selectedCountry.length ||
+    this.filter.selectedFlag.length ||
+    this.filter.selectedAttendance.length ||
+    this.filter.selectedScop
   )
     return true
   else return false
 }
 
-async function mounted (){
-  this.meetingsOptions = await getMajorMinorMeetings(this)
+async function organizationSearchChangeEvent (query){
+  this.isLoadingOrganization = true
+  if (query)
+    this.organizationOptions = await this.$kronosApi.getOrganizations({
+      FreeText: query,
+      limit   : 25
+    })
 
-  this.organizationOptions = await this.$kronosApi.getOrganizations({
-    limit: 5000
-  })
+  this.isLoadingOrganization = false
+}
+
+async function setQueryStringFilters (qs){
+  if (!_.isEmpty(qs)){
+    if (qs.FreeText) this.searchTagInput = qs.FreeText
+    selectSelectedCountry.call(this, qs.Governments)
+    selectSelectedMeetings.call(this, qs.EventUIDs)
+    selectCountryScope.call(this, qs.CountryScope)
+    selectRegistrationStatus.call(this, qs.EventRegistrationStatus)
+    await selectOrganizations.call(this, qs.OrganizationUIDs)
+  }
+}
 
-  const country = await this.$kronosApi.getCountries()
+async function selectOrganizations (organizationUIDs){
+  if (organizationUIDs){
+    this.organizationOptions = await this.$kronosApi.getOrganizations({
+      OrganizationUIDs: asArray(organizationUIDs)
+    })
+    this.filter.selectedOrganization = asArray(organizationUIDs).map(o =>
+      this.organizationOptions.find(x => x.OrganizationUID === o)
+    )
+  }
+}
 
-  sortCountryByAsc(country, this)
+function selectRegistrationStatus (regStatus){
+  if (regStatus)
+    this.filter.selectedAttendance = asArray(regStatus).map(a =>
+      this.attendanceOptions.find(x => x.value === Number(a))
+    )
+}
 
-  this.organizationTypesOptions = await this.$kronosApi.getOrganizationTypes()
+function selectCountryScope (countryScope){
+  if (countryScope)
+    this.filter.selectedScop = this.scopeOptions.find(
+      x => x.value === countryScope
+    )
 }
 
-async function getMajorMinorMeetings (_this){
+function selectSelectedCountry (countryCode){
+  if (countryCode)
+    this.filter.selectedCountry = asArray(countryCode).map(o =>
+      this.countryList.find(x => x.Code === o)
+    )
+}
+
+function selectSelectedMeetings (selectedMeetings){
+  if (selectedMeetings)
+    this.filter.selectedMeetings = asArray(selectedMeetings).map(e =>
+      this.meetingsOptions.find(x => x.EventUID === e)
+    )
+}
+
+function asArray (data){
+  return _.flatten([ data ])
```

Comment:
you can add `_compact()`. with compact falsy value are excluded from the result.

remember that 0, false, '', null, undefined., NaN are falsy values 

```suggestion
  return _([ data ]).flatten().compact().value()
```